### ISSUE TITLE
Jetstream pull subscription fetch returning empty msg

### ISSUE THREAD

**Issue by blainejohnson17** [nats-io/nats-pure.rb#91]
I'm seeing the following:

Steps:
1. create a stream and consumer
2. create a pull subscription for the consumer
(No messages exist in stream.)
3. perform fetch with the pull subscription
    -> timeout exception is raise (because no msgs are in stream (expected))
4. perform another fetch 
    -> returns a single empty msg (not expected)

This pattern repeats indefinitely (fetch => timeout, fetch => empty NATS::Msg, ....)

```
sub = nats_client.jetstream.pull_subscribe(subcription_subject, consumer_name)

sub.fetch
*** NATS::IO::Timeout Exception: nats: fetch timeout

sub.fetch
=> [#<NATS::Msg(subject: "_INBOX.NFp40anEUTX0yZqmnIdIyB", reply: "", data: "", header={"Nats-Pending-Messages"=>"1", "Nats-Pending-Bytes"=>"0"})>]
```

More Context:
**Stream Info:**
```
Configuration:

             Subjects: STREAM.SUBJECT
     Acknowledgements: true
            Retention: File - Limits
             Replicas: 1
       Discard Policy: Old
     Duplicate Window: 2m0s
    Allows Msg Delete: true
         Allows Purge: true
       Allows Rollups: false
     Maximum Messages: unlimited
        Maximum Bytes: unlimited
          Maximum Age: unlimited
 Maximum Message Size: unlimited
    Maximum Consumers: unlimited


State:

             Messages: 0
                Bytes: 0 B
             FirstSeq: 0
              LastSeq: 0
     Active Consumers: 1
```

**Consumer Info:**
```
Configuration:

        Durable Name: CONSUMER_NAME
           Pull Mode: true
      Deliver Policy: All
          Ack Policy: Explicit
            Ack Wait: 30s
       Replay Policy: Instant
     Max Ack Pending: 1,000
   Max Waiting Pulls: 512

State:

   Last Delivered Message: Consumer sequence: 0 Stream sequence: 0
     Acknowledgment floor: Consumer sequence: 0 Stream sequence: 0
         Outstanding Acks: 0 out of maximum 1,000
     Redelivered Messages: 0
     Unprocessed Messages: 0
            Waiting Pulls: 0 of maximum 512
```

**Comment by sradu** [comment]
+1 seeing the same thing

**Comment by wallyqs** [comment]
Fixed via #93
